src/components/legend/index.js

```diff
@@ -663,14 +686,53 @@ legend.repositionLegend = function(td, traces){
     lx = Math.round(lx);
     ly = Math.round(ly);
 
-    fullLayout._infolayer.selectAll('svg.legend')
-        .call(Plotly.Drawing.setRect, lx, ly, legendwidth, legendheight);
-    fullLayout._infolayer.selectAll('svg.legend .bg')
-        .call(Plotly.Drawing.setRect, borderwidth/2, borderwidth/2,
-            legendwidth-borderwidth, legendheight-borderwidth);
+
+    var legendsvg = fullLayout._infolayer.selectAll('svg.legend'),
```

Comment:
Mainly for ease of development. The scroll handler requires the final height of the legend to calculate the viewbox offset and at the time, I was still trying to wrap my head around the control flow in here. 

It will be moved out into `draw` before we merge though